packages/amplify-cli/src/__tests__/commands/init.test.ts

```diff
@@ -0,0 +1,156 @@
+import { execSync } from 'child_process';
+
+import { $TSContext, UnknownArgumentError } from 'amplify-cli-core';
+
+import { preInitSetup } from '../../init-steps/preInitSetup'
+import { analyzeProject } from '../../init-steps/s0-analyzeProject'
+import { initFrontend } from '../../init-steps/s1-initFrontend'
+import { scaffoldProjectHeadless } from '../../init-steps/s8-scaffoldHeadless';
+
+jest.mock('child_process', () => ({ execSync: jest.fn() }));
+jest.mock('fs-extra', () => ({ 
+  readdirSync: () => [],
+  copy: jest.fn(),
+  ensureDirSync: jest.fn(),
+  ensureDir: jest.fn(() => Promise.resolve()),
+  writeFileSync: jest.fn(),
+  existsSync: jest.fn(),
+  writeJSON: jest.fn(),
+  readJSON: jest.fn(() => ({})),
+}));
```

Comment:
can just do `jest.mock('package-name')` and it will attach mock functions to everything
For functions where you need a custom implementation such as readJSON, you can do:
```ts
import { readJSON } from 'fs-extra';
jest.mock('fs-extra');
readJSON_mock = readJSON as jest.Mocked<typeof readJSON>;
readJSON_mock.mockResolvedValue({});
```